tests/convention_test.py

```diff
@@ -0,0 +1,74 @@
+"""Unit tests for the metadata block in project README files."""
+
+from pathlib import Path
+
+import update_projects_table
+
+import metadata_definitions
+import pytest
+
+
+ROOT_PATH = Path(__file__).parent.parent
+
+TEST_CATEGORIES = [
+    (Path(ROOT_PATH / "devops"), "DevOps"),
+    (Path(ROOT_PATH / "reliability"), "Reliability"),
+    (Path(ROOT_PATH / "samples"), "Samples"),
+]
+
+METADATA_VALID = [
+    (metadata_definitions.CATEGORIES, "categories"),
+    (metadata_definitions.INTEGRATIONS, "integrations"),
+]
+
+
+def test_metadata_block_exists():
+    """Checks for required metadata in README files."""
+    for readme_file in sorted(Path(ROOT_PATH).rglob("README.md")):
+        metadata = update_projects_table.extract_metadata(readme_file)
```

Comment:
Notice that all your tests (unsurprisingly) start with the same code:
```py
    for readme_file in sorted(Path(ROOT_PATH).rglob("README.md")):
        metadata = update_projects_table.extract_metadata(readme_file)
```
You can create a helper function that returns (or even better - yields) `readme_file` and `metadata`.

...And you can also skip files that don't have metadata.